internal/provider/waypoint/resource_waypoint_application_template.go

```diff
@@ -361,6 +423,28 @@ func (r *ApplicationTemplateResource) Read(ctx context.Context, req resource.Rea
 		data.TerraformNoCodeModule = tfcNoCode
 	}
 
+	if appTemplate.VariableOptions != nil && len(appTemplate.VariableOptions) > 0 {
+		varOpts := []*tfcVariableOption{}
+		for _, v := range appTemplate.VariableOptions {
+			varOptsState := &tfcVariableOption{
+				Name:         types.StringValue(v.Name),
+				VariableType: types.StringValue(v.VariableType),
+			}
+
+			vOpts, diags := types.ListValueFrom(ctx, types.StringType, v.Options)
```

Comment:
This is slightly different than what you're doing on line 351, might be worth making a helper function for reading variable options so that it's the same everywhere you do it.

Reply:
I added the helper `readVarOpts` [here](https://github.com/hashicorp/terraform-provider-hcp/blob/454a6b3469c16d5e7865dc3bad838e79a3131f8d/internal/provider/waypoint/resource_waypoint_application_template.go#L367-L393). This is used by both the data source and the resource.